stats-test.cpp

```diff
@@ -17,7 +18,9 @@ TEST_CASE("average is NaN for empty array") {
     auto computedStats = Statistics::ComputeStatistics({});
     //All fields of computedStats (average, max, min) must be
     //NAN (not-a-number), as defined in math.h
-    
+    REQUIRE(std::isnan(computedStats.average));
+    REQUIRE(std::isnan(computedStats.max));
+    REQUIRE(std::isnan(computedStats.min));
```

Comment:
This test is correct. Extend the tests here. What result do you want if the input itself contains NaNs? Write a test first, design the REQUIRE statement and then fix if needed.

Reply:
Clear. The test clearly expresses that we don't tolerate NaNs in the input.